### PR TITLE
Add coffee build output for non-coffee module users

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: minor
Connects-to: #1
Signed-off-by: Cameron Diver <cameron@resin.io>

### CODE REVIEW

**Review by hedss — CHANGES_REQUESTED** [review]
What we usually do for modules written in CoffeeScript that need to be general accessible is to build them with `gulp`, with appropriate instructions in the `package.json` for prepublishing/build/test instructions.

That way you can have the `main: lib.js` property set and know that you don't have to do everything yourself (esp. with prepublish).

Here's a couple of repos that use `gulp`:
https://github.com/resin-io/resin-cli
https://github.com/resin-io-modules/resin-discoverable-services

And here's a link to the NPM page describing `prepublish` and building for other languages (handily it includes CS!):
https://docs.npmjs.com/files/package.json

**Review by hedss — APPROVED** [review]
LGTM!

File: package.json [hunk 1]
```diff
@@ -11,9 +11,9 @@
   "author": "Aleksis Brezas <abresas@resin.io>",
   "license": "Apache-2.0",
   "scripts": {
-    "pretest": "./node_modules/.bin/coffeelint lib.coffee dockerfile-template.coffee ./test",
-    "test": "./node_modules/mocha/bin/mocha --compilers coffee:coffee-script/register",
-    "prepublish": "./node_modules/.bin/coffee -c lib.coffee dockerfile-template.coffee && (echo '#!/usr/bin/env node' | cat - dockerfile-template.js > dockerfile-template)"
+    "pretest": "coffeelint lib.coffee dockerfile-template.coffee ./test",
```
Comment (Page-) [inline package.json:14]: Can you switch to resin-lint please :)

File: package.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
...
+    "test": "mocha --compilers coffee:coffee-script/register",
+    "prepublish": "gulp && (echo '#!/usr/bin/env node' | cat - dockerfile-template.js > dockerfile-template)"
```
Comment (Page-) [inline package.json:16]: This should use the [`bin`](https://docs.npmjs.com/files/package.json#bin) property of package.json in order to make the script executable - it handles the execute permission bit on linux, and also generates a .cmd for windows

**Pull request merged**